apis/core/v1alpha1/resource.go

```diff
@@ -124,10 +158,15 @@ type ResourceStatus struct {
 	BindingStatus     `json:",inline"`
 }
 
-// NonPortableClassSpecTemplate defines a template that will be used to create
-// the specifications of managed resources dynamically provisioned using a
-// resource class.
-type NonPortableClassSpecTemplate struct {
+// A ClassSpecTemplate defines a template that will be used to create the
+// specifications of managed resources dynamically provisioned using a resource
+// class.
+type ClassSpecTemplate struct {
+	// WriteConnectionSecretsToNamespace specifies the namespace in which the
+	// connection secrets of managed resources dynamically provisioned using
+	// this claim will be created.
+	WriteConnectionSecretsToNamespace string `json:"writeConnectionSecretsToNamespace"`
```

Comment:
Is this the remedy to the problem proposed in this slack message: https://crossplane.slack.com/archives/CKXQHM7U3/p1571426744220400 ? If so, I like it! I appreciate how it allows the user to choose the secret's namespace, even if we may recommend that they usually choose the same namespace as the `Provider` secret.

Reply:
Yeah, this was option two proposed in that discussion. I went with this rather than option 3 because it seemed simpler, more flexible, and more explicit/less magic. I agree that in practice we probably just want to recommend all providers and classes use `crossplane-system` for their secrets.